src/main/java/com/epam/test_generator/services/TokenService.java

```diff
@@ -43,16 +46,25 @@ public String getToken(LoginUserDTO userDTO) {
 
         if (user == null) {
             throw new UnauthorizedException(
-                "User with email: " + userDTO.getEmail() + " not found.");
+                    "User with email: " + userDTO.getEmail() + " not found.");
+        }
+
+        if (user.isLocked()) {
+            throw new UnauthorizedException("User Account is locked!");
         }
 
         if (!(userService.isSamePasswords(userDTO.getPassword(), user.getPassword()))) {
-            throw new UnauthorizedException("Incorrect password.");
+            int attempts = userService.updateFailureAttempts(user.getId());
+            if (user.isLocked()) {
+                throw new UnauthorizedException("Incorrect password. User is locked!");
+            }
+            throw new UnauthorizedException("Incorrect password. Attempts: " + attempts);
         } else {
             Builder builder = JWT.create()
-                .withIssuer("cucumber")
-                .withClaim("id", user.getId());
+                    .withIssuer("cucumber")
+                    .withClaim("id", user.getId());
             try {
+                userService.invalidateAttempts(user.getId());
                 return builder.sign(Algorithm.HMAC256(environment.getProperty("jwt_secret")));
             } catch (UnsupportedEncodingException e) {
                 throw new RuntimeException(e.getMessage());
```

Comment:
It will be better to refactor this method with extract private methods, because it's unreadable.